### PR TITLE
123 unexpected use of tmp path

### PR DESCRIPTION
**Pull request by markcurtis1970**
Re-opened original PR made corrective changes and merged back in main.

### CODE REVIEW

File: cmd/local/conf/conf.go [hunk 1]
```diff
@@ -354,9 +354,12 @@ func ReadConf(overrides map[string]string, configDir string) (*CollectConf, erro
 	return c, nil
 }
 
-func getOutputDir(now time.Time) string {
-	nowStr := now.Format("20060102-150405")
-	return filepath.Join(os.TempDir(), "ddc", nowStr)
+func getOutputDir(outputDir string, t time.Time) string {
+	nowStr := t.Format("20060102-150405")
+	if outputDir == "" {
+		return filepath.Join(os.TempDir(), "ddc", nowStr)
```
Comment (rsvihladremio) [inline cmd/local/conf/conf.go:360]: need to remove call to TempDir here

Comment (markcurtis1970) [reply cmd/local/conf/conf.go:360]: Actually thats a "safety net" if the "output dir" is empty we have to have a value in here

File: cmd/local/conf/conf.go [hunk 2]
```diff
@@ -167,15 +167,21 @@ func ReadConf(overrides map[string]string, configDir string) (*CollectConf, erro
...
+	c := &CollectConf{}
 	// set node name
 	hostName, err := os.Hostname()
 	if err != nil {
 		hostName = fmt.Sprintf("unknown-%v", uuid.New())
 	}
+	//set default tmp dir (if its not in the conifg, use the OS default)
+	c.outputDir = GetString(confData, KeyTmpOutputDir)
+	if c.outputDir == "" {
+		c.outputDir = filepath.Join(os.TempDir(), getOutputDir(time.Now()))
```
Comment (rsvihladremio) [inline cmd/local/conf/conf.go:180]: ok the thing negative with this is, this means that the "standard directory name structure" of ddc gets lost if the output dir is set, when really we just need to not use the temp dir when the output dir is set

**Comment by rsvihladremio** [comment]
closing for now, the code has changed a lot and we will need a revisit unfortunately

**Pull request closed without merging**